modules/generator/processor/hostinfo/config.go

```diff
@@ -0,0 +1,21 @@
+package hostinfo
+
+import (
+	"flag"
+	"time"
+)
+
+type Config struct {
+	// HostIdentifiers defines the list of resource attributes used to derive
+	// a unique `grafana.host.id` value. In most cases, this should be [ "host.id" ]
+	HostIdentifiers []string      `yaml:"host_identifiers"`
+	StaleDuration   time.Duration `yaml:"stale_duration"`
+	// MetricName defines the name of the metric that will be generated
+	MetricName string `yaml:"metric_name"`
+}
+
+func (cfg *Config) RegisterFlagsAndApplyDefaults(string, *flag.FlagSet) {
+	cfg.HostIdentifiers = []string{"host.id"}
+	cfg.StaleDuration = 2 * time.Minute
```

Comment:
Attempting to set the default stale duration as low as possible here as we depend on the absence of these metrics to get accurate counts. I noticed the ManagedRegistry only removes stale series on a 5 minute interval though, so this might be confusing if set to anything less than 5 minutes.

https://github.com/grafana/tempo/blob/main/modules/generator/registry/registry.go#L134

Reply:
Thanks Joe, makes sense!